json_serializable/test/integration/json_test_example.g_any_map.g.dart

```diff
@@ -85,26 +85,17 @@ Order _$OrderFromJson(Map json) {
         StatusCode.success;
 }
 
-Map<String, dynamic> _$OrderToJson(Order instance) {
-  final val = <String, dynamic>{};
-
-  void writeNotNull(String key, dynamic value) {
-    if (value != null) {
-      val[key] = value;
-    }
-  }
-
-  writeNotNull('count', instance.count);
-  val['isRushed'] = instance.isRushed;
-  val['duration'] = instance.duration?.inMicroseconds;
-  val['category'] = _$CategoryEnumMap[instance.category];
-  val['items'] = instance.items;
-  val['platform'] = instance.platform;
-  val['altPlatforms'] = instance.altPlatforms;
-  val['homepage'] = instance.homepage?.toString();
-  val['status_code'] = _$StatusCodeEnumMap[instance.statusCode];
-  return val;
-}
+Map<String, dynamic> _$OrderToJson(Order instance) => <String, dynamic>{
+      if (instance.count case final val?) 'count': val,
```

Comment:
On the flip-side (I can't comment there), it would be great for final code size to have a library of helpers for common patterns like
```dart
..count = (json['count'] as num?)?.toInt()
```
-->
```dart
..count = helper$.nullableInt(json, 'count')
```

Reply:
How could we measure the delta here? I'm game, but I'd like to see numbers...

I wonder what the lint is here, too. If this is such a big win...how can we codify it more?

Reply:
If we land parts-with-imports this gets easy enough to just do it without worrying about the specific delta.

https://github.com/dart-lang/language/blob/main/working/augmentation-libraries/parts_with_imports.md

IMO it's not worth spending much time one before then.